services/api-server/src/simcore_service_api_server/services/log_streaming.py

```diff
@@ -137,26 +136,19 @@ async def __aexit__(self, exc_type, exc, tb):
 
     async def _project_done(self) -> bool:
         task = await self._director2_api.get_computation(self._job_id, self._user_id)
-        return not task.stopped is None
+        return not (task.stopped is None)
 
     async def log_generator(self) -> AsyncIterable[str]:
         if not self._is_registered:
             raise LogStreamerNotRegistered(
                 f"LogStreamer for job_id={self._job_id} is not correctly registered"
             )
-        last_log_time: datetime | None = None
         while True:
-            while self._queue.empty():
+            try:
+                log: JobLog = await asyncio.wait_for(
+                    self._queue.get(), timeout=self._log_check_timeout
+                )
+                yield log.json() + _NEW_LINE
+            except asyncio.TimeoutError:
                 if await self._project_done():
```

Comment:
MINOR: I think it would be more readable as

```python
done = False

while not done:
         try:
              #...
         except asyncio.TimeoutError:
              done = await self._project_done()

Reply:
Good point. I will fix